Makefile.am

```diff
@@ -204,6 +204,34 @@ DISTCLEANFILES += $(polkit_policy_DATA)
 
 ###############################################################################
 
+check_programs += shared/nm-utils/tests/test-shared-general
+
+shared_nm_utils_tests_test_shared_general_CPPFLAGS = \
+	$(dflt_cppflags) \
+	-I$(srcdir)/shared \
+	-DNETWORKMANAGER_COMPILATION_TEST \
+	-DNETWORKMANAGER_COMPILATION='(NM_NETWORKMANAGER_COMPILATION_GLIB|NM_NETWORKMANAGER_COMPILATION_WITH_GLIB_I18N_PROG)' \
+	$(CODE_COVERAGE_CFLAGS) \
+	$(GLIB_CFLAGS) \
+	$(LIBUDEV_CFLAGS) \
+	$(SANITIZER_LIB_CFLAGS) \
+	$(NULL)
+
+shared_nm_utils_tests_test_shared_general_SOURCES = \
+	shared/nm-utils/nm-hash-utils.c \
+	shared/nm-utils/nm-random-utils.c \
+	shared/nm-utils/nm-shared-utils.c \
+	shared/nm-utils/nm-time-utils.c \
+	shared/nm-utils/tests/test-shared-general.c \
+	$(NULL)
+
+shared_nm_utils_tests_test_shared_general_LDADD = \
+	$(GLIB_LIBS) \
+	shared/libcsiphash.la \
+	$(NULL)
+
+###############################################################################
+
```

Comment:
Can you also add this test to meson?

Reply:
Done. How about now?

Yes, we currently build the same source files multiple times. That is wrong, we should build intermedate static libraries. However, the reason is, that currently we pass different CFLAGS depending on where the shared files are used. So, cleaning this up, is more effort (and beyond the point of this branch). TODO.

Reply:
LGTM